src/i18n/request.ts

```diff
@@ -0,0 +1,17 @@
+import { hasLocale } from 'next-intl';
+import { getRequestConfig } from 'next-intl/server';
+
+import { routing } from '@/i18n/routing';
+
+export default getRequestConfig(async ({ requestLocale }) => {
+  const requested = await requestLocale;
+  const locale = hasLocale(routing.locales, requested)
+    ? requested
+    : routing.defaultLocale;
+
+  return {
+    locale,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+    messages: (await import(`../../messages/${locale}.json`)).default,
+  };
+});
```

Comment:
It is quite expected that without the disable directive, there will be a linter error here, because TypeScript cannot predict what exactly will be inside the JSON. I think that using the disable directive is appropriate here. The following workaround also worked for me:
```suggestion
    return {
    locale,
    messages: (
      (await import(`../../messages/${locale}.json`)) as { default: object }
    ).default,
  };
```

Reply:
excellent solution, thanks ))